### DISCUSSION TITLE
Preserving grain morphology when refining grains with lamellar

### DISCUSSION THREAD

**Discussion by mario-renderos** [neperfepx/neper#954] [Q&A] [answered]
Hello, good morning.

I have created a 2d tessellation (in a circle domain), where I have refined certain specific grains.
![z_view_cil_3d_v3_refined_corte](https://github.com/user-attachments/assets/20f3211c-46b7-4aa3-90e5-cca5a360bf3c)
When I trying to create a lamellar refinement of another grains from the *.tess file (taking my 2d tessellation file as the initial morphology), I am unable to preserve the morphology of the refined grains along with the new grains in lamellar.

`neper -T -n 59::from_morpho -morpho "voronoi::lamellar(n=file(n_div_recorte),v=file(normales_recorte))" \
 -morphooptiini "file(cil_3d_v3_refined_corte.tess)" -domain "cylinder(0.005,0.4,80)" -o cil_3d_v3_refined_laminated`

`neper -V cil_3d_v3_refined_laminated.tess -datacellcol id -cameracoo x:y:1.25 -camerasky 0:1:0 -datacelltrs 0.1 -print z_view_cil_3d_v3_refined_laminated`

![z_view_cil_3d_v3_refinado_laminado](https://github.com/user-attachments/assets/8c0fecb5-c64c-4df7-9d40-7d20aab18281)

I would like to know how to do the lamellar refinement while preserving the morphology of the previous grains based on the original 2d *.tess file.

Thank you very much in advance.

**Comment by rquey** [comment, marked as answer]
I think it is not possible.  You have to do all at once, always starting from the 1st-scale tessellation, and combining the information of the 2nd scale.

**Reply by mario-renderos** [reply]
Thank you very much, I will try to use the two-scale strategy at once.
